db/python/tables/project.py

```diff
@@ -465,209 +266,71 @@ async def delete_project_data(
+            """,
+                {'project_id': project.id},
             )
-        return bool(value)
-
-    async def check_if_member_in_group_name(self, group_name: str, member: str) -> bool:
-        """Check if a member is in a group"""
-        if self.allow_full_access:
-            return True
 
-        _query = """
-            SELECT COUNT(*) > 0
-            FROM group_member gm
-            INNER JOIN `group` g ON g.id = gm.group_id
-            WHERE g.name = :group_name
-            AND gm.member = :member
-        """
-        value = await self.connection.fetch_val(
-            _query, {'group_name': group_name, 'member': member}
-        )
-        if value not in (0, 1):
-            raise ValueError(
-                f'Unexpected value {value!r} when determining access to {group_name} '
-                f'for {member}'
+            new_audit_log_id = await self.audit_log_id()
+
+            db_members: list[dict[str, str]] = []
+
+            for m in members:
+                db_members.extend([{'member': m.member, 'role': r} for r in m.roles])
+
+            await self.connection.execute_many(
+                """
+                    INSERT INTO project_member
+                        (project_id, member, role, audit_log_id)
+                    VALUES (:project_id, :member, :role, :audit_log_id);
+                """,
+                [
+                    {
+                        'project_id': project.id,
+                        'member': m['member'],
+                        'role': m['role'],
+                        'audit_log_id': audit_log_id_map.get(
+                            (m['member'], m['role']), new_audit_log_id
+                        ),
+                    }
+                    for m in db_members
+                    if m['role'] in project_member_role_names
+                ],
             )
 
-        return bool(value)
-
-    async def check_which_groups_member_has(
-        self, group_ids: set[int], member: str
-    ) -> set[int]:
-        """
-        Check which groups a member has
-        """
-        if self.allow_full_access:
-            return group_ids
-
-        _query = """
-            SELECT gm.group_id as gid
-            FROM group_member gm
-            WHERE gm.member = :member AND gm.group_id IN :group_ids
-        """
-        results = await self.connection.fetch_all(
-            _query, {'group_ids': group_ids, 'member': member}
-        )
-        return set(r['gid'] for r in results)
-
-    async def create_group(self, name: str, audit_log_id: int) -> int:
-        """Create a new group"""
-        _query = """
-            INSERT INTO `group` (name, audit_log_id)
-            VALUES (:name, :audit_log_id)
-            RETURNING id
-        """
-        return await self.connection.fetch_val(
-            _query, {'name': name, 'audit_log_id': audit_log_id}
-        )
+        if self._connection:
+            await self._connection.refresh_projects()
```

Comment:
This is pretty circular haha, but makes sense that we're caching this information on the connection now

Reply:
Yeah this is really only for the tests, seeing as the connection is constructed per-request it generally isn't necessary to refresh them. It's only the tests where we create a project and then do something else